nix-libs/nodeLib/tools/execute-install-scripts

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+import json, subprocess, os, sys
+
+# In case a build step references the home directory, set it here.
+if not os.getenv("HOME"):
+   os.environ["HOME"] = os.getcwd()
+
+scripts = json.load(open("package.json")).setdefault("scripts", {})
+
+def runscript(name):
+    if scripts.get(name):
+        print("Running {} defined in package.json".format(repr(name)))
+        sys.stdout.flush()
+        proc = subprocess.Popen(["/bin/sh", "-x"], stdin=subprocess.PIPE)
```

Comment:
Is there anyway we can generate this script so that we  can interpolate the shell path?

Reply:
I changed the definition of `node-build-tools` to add a wrapper around this script which sets the PATH to nix store paths, so that all of these binaries are deterministic. We could do interpolation instead, but I like having the script be at least runnable in its raw form